### DISCUSSION TITLE
ignore workspace if it doesn't exist.

### DISCUSSION THREAD

**Discussion by azinsharaf** [epwalsh/obsidian.nvim#443] [Q&A] [answered]
hi, i use one neovim and `obsidian.lua` config file on both home and work computers. I have defined two workspaces `iCloud` (home) and `OneDrive` (work). I can't install iCloud drive on work computer so `obsidian.nvim` complains if i open nvim at work. can i configure it in a way that if the workspace folder doesn't exist it just skips that without `mkdir error`?

**Comment by epwalsh** [comment, marked as answer]
Hey @azinsharaf, I would recommend dynamically setting your workspace config. For example:

```lua
local Path = require("plenary.path")

local workspaces = {}

if Path:new("/path/to/onedrive/workspace"):exists() then
  table.insert(workspaces, { path = "...", ... })
end

require("obsidian").setup({ workspaces = workspaces, ... })
```

**Reply by azinsharaf** [reply]
thanks. I have added them the way you mentioned (with some more tweaks). I can switch between them successfully. i can create today's note successfully too. but two issues:
- `:Obsidianquickswitch` shows an empty result window. 
- `:ObsidianCheck` returns this error:

```
Error executing luv callback:
.../nvim-data/lazy/plenary.nvim/lua/plenary/async/async.lua:18: The coroutine failed with this message: ...-data/lazy/obsidian.nvim/lua/obsidian/commands/check.lua:45: attempt to concatenate local 'relative_path' (a table value)                                                                                                                                    stack traceback:                                                                                                                                                                            [C]: in function 'error'                                                                                                                                                            .../nvim-data/lazy/plenary.nvim/lua/plenary/async/async.lua:18: in function 'callback_or_next'                                                                                      .../nvim-data/lazy/plenary.nvim/lua/plenary/async/async.lua:45: in function <.../nvim-data/lazy/plenary.nvim/lua/plenary/async/async.lua:44>                                Press ENTER or type command to continue
```
my config file: 

```
local Path = require("plenary.path")

local workspacePaths = {
	{ name = "work", path = "C:/Users/asharaf/OneDrive - CompanyName/Obsidian/Obsidian_work" },
	{ name = "personal", path = "C:/Users/asharaf/iCloudDrive/iCloud~md~obsidian/obsidian_personal" },
}

local workspaces = {}

for _, workspaceInfo in ipairs(workspacePaths) do
	local workspacePath = workspaceInfo.path
	if Path:new(workspacePath):exists() then
		local workspace = {
			name = workspaceInfo.name,
			path = workspacePath,
		}
		table.insert(workspaces, workspace)
	end
end

return {
	"epwalsh/obsidian.nvim",
	version = "*", -- recommended, use latest release instead of latest commit
	lazy = false,
	ft = "markdown",
	-- Replace the above line with this if you only want to load obsidian.nvim for markdown files in your vault:
	-- event = {
	--   -- If you want to use the home shortcut '~' here you need to call 'vim.fn.expand'.
	--   -- E.g. "BufReadPre " .. vim.fn.expand "~" .. "/my-vault/**.md"
	--   "BufReadPre path/to/my-vault/**.md",
	--   "BufNewFile path/to/my-vault/**.md",
	-- },
	dependencies = {
		-- Required.
		"nvim-lua/plenary.nvim",

		-- see below for full list of optional dependencies 👇
	},
	opts = {
		workspaces = workspaces,

		-- see below for full list of options 👇
	},
}
```

**Reply by epwalsh** [reply]
@azinsharaf can you make sure you have the latest commit of obsidian.nvim and then run those commands again. If you're still getting an error, please run `:ObsidianDebug`.

**Reply by epwalsh** [reply]
I did just find a bug with `:ObsidianCheck`. Just pushed the fix to `main`.

**Reply by azinsharaf** [reply]
ok, i am on `3.6.1` version. :`Obsidinacheck` works fine now. but `quickswitch` still returns nothing. 
`:ObsidianDebug` result:

![image](https://github.com/epwalsh/obsidian.nvim/assets/10101468/74b2328a-448f-471c-b435-aec174d60b6e)

**Reply by epwalsh** [reply]
That all looks fine to me. When you run the quick switch command are there any warnings/errors in `:messages`? If I had to guess, I'd say you should see something like "error: no picker configured".

**Reply by azinsharaf** [reply]
omg! very smart. I had turned off messages so running the command shows `rg` executable not found.

**Reply by epwalsh** [reply]
`ripgrep` (`rg`) is a lot more versatile than `fzf` as we can use it for both content and filename searches. It's also compatible with every major OS, is easy to install, and is incredibly fast.

It's nice that telescope supports using different backend tools like `fzf`, but that's outside the scope for obsidian.nvim.

**Reply by azinsharaf** [reply]
installing rg fixed it.

**Reply by xarthurx** [reply]
@epwalsh 

Is it possible to use global variables for the path dir? I cannot make it correct after a few attempts...

https://github.com/epwalsh/obsidian.nvim/discussions/536
